### ISSUE TITLE
NodeJS running out of memory while reading small file

### ISSUE THREAD

**Issue by yashgt** [guyonroche/exceljs#133]
Hi,

I am reading a workbook in memory using:

```
	var wb = new Excel.Workbook();
	wb.xlsx.readFile(filename).then(function() {});
```
		
The size of the files as shown by ls -l is:
8895940 Jun  9 22:06 3-TimeTable.xlsx

That is close to 9MB.

The code works fine on my laptop on Win 7 with 4GB RAM.

However, on Amazon EC2 Micro instance I get the following error:

> <--- Last few GCs --->
> 
>    99229 ms: Mark-sweep 675.9 (4906.0) -> 675.9 (7569.0) MB, 541.9 / 0 ms [allocation failure] [GC in old space requested].
>    99778 ms: Mark-sweep 675.9 (7570.0) -> 675.9 (10230.0) MB, 549.2 / 0 ms [allocation failure] [GC in old space requested].
>   100346 ms: Mark-sweep 675.9 (10231.0) -> 666.9 (11997.0) MB, 567.4 / 0 ms [last resort gc].
>   100910 ms: Mark-sweep 666.9 (11997.0) -> 667.1 (12796.0) MB, 563.8 / 0 ms [last resort gc].
> 
> 
> <--- JS stacktrace --->
> 
> ==== JS stack trace =========================================
> 
> Security context: 0x3d06ce9b4629 <JS Object>
>     1: DELETE [native runtime.js:350] [pc=0x394d59b8a51] (this=0x1d62288229c9 <an Object with map 0x11ac6a384481>,q=0x2e5393f075f9 <String[1]: s>,r=1)
>     2: /* anonymous */(aka /* anonymous */) [/home/ec2-user/raloa1/code/admin/node_modules/exceljs/lib/xlsx/xlsx.js:740] [pc=0x394d59a605b] (this=0x3d06ce9041b9 <undefined>,cell=0x1d62288229c9 <an Object with map 0x11ac6a384481>)
>     3: arguments...
> 
> FATAL ERROR: Committing semi space failed. Allocation failed - process out of memory
> Aborted

The Micro instance has 1 GB RAM.

> [ec2-user@ipadmin]$ free -m
>               total        used        free      shared  buff/cache   available
> Mem:            991          69         793          56         128         761
> Swap:             0           0           0
> 

761 MB seems to be available.

Why would reading an 8MB file run out of memory?

Is there some node.js option I should be using to get it working?

Thanks

**Comment by mfahmirukman** [comment]
I'm facing the same issue but not with the amazon server.

I can read/write large sized file on my laptop/pc, but when server handles it, it'd fail.

Your problem is that exceljs doesn't exactly release memory after reading/writing and I'm still trying to find out if there's a workaround to this problem

**Comment by andresgottlieb** [comment]
@Siemienik it's still happening to me with v1.14.0. A 10MB file eats +2GB of RAM and my server crashes. I had to move to [SheetJS](https://github.com/SheetJS/js-xlsx), and it worked like a charm (around 500MB RAM)

**Comment by shubham-marathia** [comment]
Yes, exceljs isnt releasing memory while writing to excel. I have optimised my program to a very high level, still facing the issue.

**Comment by anishbishnoi127** [comment]
same issue anyone foud solution ? please let me know.
my excel file containg the formating also and excel file size is 4mb and im not able to load that file into the nodejs it;s doing heap memory issue.
i just want to open that file in nodejs and want to particular sheet inside put the data also want to keep the other sheet styling keep it . anyone know how do this ?
